components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -103,39 +177,35 @@ def search_and_schedule_new_tasks(db_conn, db_cursor, clp_metadata_db_connection
             clp_metadata_db_connection_config=clp_metadata_db_connection_config,
         )
 
-        for path_idx, path in enumerate(clp_io_config.input.paths_to_compress, start=1):
-            path = Path(path)
-
-            try:
-                file, empty_directory = validate_path_and_get_info(
-                    CONTAINER_INPUT_LOGS_ROOT_DIR, path
+        input_config = clp_io_config.input
+        input_type = input_config.type
+        if input_type == "fs":
+            _process_fs_input_paths(input_config, paths_to_compress_buffer)
+        elif input_type == "s3":
```

Comment:
```suggestion
        if input_type == InputType.FS.value:
            _process_fs_input_paths(input_config, paths_to_compress_buffer)
        elif input_type == InputType.S3.value:
```

Unless this doesn't work for some reason.